### PR TITLE
Prevents players from exploting the pricetag component on bounty cubes.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

So, I was convinced that this had been fixed at some point, and it would appear that either it unfixed itself, never got fixed in the first place, or got reverted at some point, so this is the official fix for this.

By wrapping up bounty cubes and placing a new pricetag on the cube, players were able to effectively overwrite the price tag built in for cargo bounties and get up to 50% payout instead of the intended cut mean for you and cargo. This was not and is not intented. Instead, this gives the pricetag component COMPONENT_DUPE_UNIQUE, preventing it from being overwritten via price tags while wrapped. The only intentional way to remove a pricetag component remains unwrapping it from a package, which bounty cubes have been made exempt from as well.

The ONLY processing you are expected to perform on a bounty cube is JUST shipping it out.

## Why It's Good For The Game

This was absolutely an unintended exploit capable of both stealing from players (Removing the pricetag for solo-cargo civilian bounties), and getting much higher civilian payouts by the time of sale (replacing player cut to 50% profit.)

## Changelog
:cl:
fix: An exploit with civilian bounty cubes, allowing you to rewrap and re-pricetag cubes for higher profits has been fixed.
/:cl:

### CODE REVIEW

File: code/datums/components/pricetag.dm [hunk 1]
```diff
@@ -19,7 +20,8 @@
 
 /datum/component/pricetag/proc/Unwrapped()
 	SIGNAL_HANDLER
-
+	if(istype(parent, /obj/item/bounty_cube))
+		return
```
Comment (TemporalOroboros) [inline code/datums/components/pricetag.dm:24]: Wouldn't it be easier to just not register this?

**Comment by Pugie** [comment]
Emergent gameplay - this is what cargo techs deserve for not ripping up package wrapping as they have no reason not to do so other than RP.

**Comment by maxymax13** [comment]
you can bugfix all you can but you still aren't fixing the fact that bounties are an useless chore that makes players go on pointless fetch quests and then annoy cargonians busy doing more important things (shocking themselves building emitters so that a newfriend techie can swap the guns crate with an abandoned crate with the same exact sprite at the last minute and blow all our limbs off) and even when do you get their 500 yellow pulsating suspiciously goon-like sprites onto the shuttle all either party gets is 5 and a half credits, a button, and pocket lint

**Comment by YakumoChen** [comment]
Cargo techs are perfectly capable of unwrapping stuff themselves if they're unhappy with players trying to get a better cut of the deal.

File: code/datums/components/pricetag.dm (hunk 1 above)
Comment (Mothblocks) [reply code/datums/components/pricetag.dm:24]: i agree, this is a hack

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/components/pricetag.dm (hunk 1 above)
Comment (Mothblocks) [reply code/datums/components/pricetag.dm:24]: more specifically, i think components should have the ability to apply to anything aside from actual technical limitations, not design ones.

the bounty cube itself should tell the price tag component not to register it

**Comment by cacogen** [comment]
There was a separate exploit I introduced earlier this year where you could wrap a bounty cube and use a pricetag to get a huge payout of like 50x I think instead of 0.3x or whatever it's supposed to be normally. I fixed that. That may be what you're thinking of.

**Pull request closed without merging**